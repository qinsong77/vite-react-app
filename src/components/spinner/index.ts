export * from './spinner'
export * from './type'
