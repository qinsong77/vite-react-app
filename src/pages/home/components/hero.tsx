import reactLogo from '@/assets/react.svg'

import PageIllustration from './page-illustration'
import viteLogo from '/vite.svg'

const ImageLogo = () => {
  return (
    <div className="mb-6 flex justify-center" data-aos="fade-up">
      <a href="https://vitejs.dev" target="_blank" rel="noreferrer">
        <img
          src={viteLogo}
          className="h-32 p-3 drop-shadow-lg transition hover:drop-shadow-2xl"
          alt="Vite logo"
        />
      </a>
      <a href="https://react.dev" target="_blank" rel="noreferrer">
        <img
          src={reactLogo}
          className="h-32 animate-spin p-3 drop-shadow-lg transition hover:drop-shadow-2xl"
          alt="React logo"
        />
      </a>
    </div>
  )
}
export default function Hero() {
  return (
    <section>
      <div className="relative mx-auto px-4 sm:px-6">
        {/* Illustration behind hero content */}
        <PageIllustration />
        {/* Hero content */}
        <div className="relative pt-16 pb-10 md:pt-20 md:pb-16">
          {/* Section header */}
          <ImageLogo />
          <div className="mx-auto max-w-3xl text-center">
            <h1 className="mb-8 text-2xl" data-aos="fade-up">
              Landing template for startups
            </h1>
            <p
              className="text-muted-foreground mb-8 text-lg italic"
              data-aos="fade-up"
              data-aos-delay="200"
            >
              Our landing page template works on all devices, so you only have
              to set it up once, and get beautiful results forever.
            </p>
            <div className="mx-auto max-w-xs sm:flex sm:max-w-none sm:justify-center">
              <div data-aos="fade-up" data-aos-delay="400">
                <a
                  className="mb-4 w-full rounded-lg bg-purple-600 px-4 py-2.5 text-white hover:bg-purple-700 sm:mb-0 sm:w-auto"
                  href="#0"
                >
                  Start free trial
                </a>
              </div>
              <div data-aos="fade-up" data-aos-delay="600">
                <a
                  className="w-full rounded-lg bg-gray-700 px-4 py-2.5 text-white hover:bg-gray-800 sm:ml-4 sm:w-auto"
                  href="#0"
                >
                  Learn more
                </a>
              </div>
            </div>
          </div>
        </div>
      </div>
    </section>
  )
}
