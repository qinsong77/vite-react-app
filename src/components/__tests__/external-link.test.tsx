import { render, screen } from '@testing-library/react'
import { userEvent } from '@testing-library/user-event'
import { expect } from 'vitest'

import { ExternalLink } from '@/components/external-link'

describe('<ExternalLink />', () => {
  it('should render with svg icon', () => {
    const href = 'https://sysuke.com'

    const { asFragment } = render(
      <ExternalLink href={href}>Sysuke</ExternalLink>
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should render with a tag', async () => {
    const user = userEvent.setup()
    const href = 'https://sysuke.com'
    render(<ExternalLink href={href}>Sysuke</ExternalLink>)
    expect(screen.getByRole('link', { name: 'Sysuke' })).toHaveAttribute(
      'href',
      href
    )
    await user.click(screen.getByRole('link', { name: 'Sysuke' }))
  })
})
