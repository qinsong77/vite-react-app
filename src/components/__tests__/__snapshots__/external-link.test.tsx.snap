// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<ExternalLink /> > should render with svg icon 1`] = `
<DocumentFragment>
  <a
    class="inline-flex flex-1 justify-center gap-1 leading-4 hover:underline"
    href="https://sysuke.com"
    rel="noreferrer"
    target="_blank"
  >
    <span>
      Sysuke
    </span>
    <svg
      aria-hidden="true"
      class="opacity-70"
      height="7"
      viewBox="0 0 6 6"
      width="7"
    >
      <path
        d="M1.25215 5.54731L0.622742 4.9179L3.78169 1.75597H1.3834L1.38936 0.890915H5.27615V4.78069H4.40513L4.41109 2.38538L1.25215 5.54731Z"
        fill="currentColor"
      />
    </svg>
  </a>
</DocumentFragment>
`;
